src/main/java/fruit/shop/service/TransactionStrategyImpl.java

```diff
@@ -0,0 +1,24 @@
+package fruit.shop.service;
+
+import fruit.shop.model.FruitTransaction;
+import fruit.shop.model.TransactionType;
+import fruit.shop.service.transaction.BalanceTransactionHandler;
+import fruit.shop.service.transaction.PurchaseTransactionHandler;
+import fruit.shop.service.transaction.ReturnTransactionHandler;
+import fruit.shop.service.transaction.SupplyTransactionHandler;
+import fruit.shop.service.transaction.TransactionHandler;
+import java.util.Map;
+
+public class TransactionStrategyImpl implements TransactionStrategy {
+    private final Map<TransactionType, TransactionHandler> transactionHandlerMap = Map.ofEntries(
+            Map.entry(TransactionType.BALANCE, new BalanceTransactionHandler()),
+            Map.entry(TransactionType.SUPPLY, new SupplyTransactionHandler()),
+            Map.entry(TransactionType.PURCHASE, new PurchaseTransactionHandler()),
+            Map.entry(TransactionType.RETURN, new ReturnTransactionHandler())
+    );
```

Comment:
Let's make this implementation a bit more flexible.

1. Make this field initialized in constructor
2. Create map and pass it to constructor in `main()` method.